rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -153,14 +154,26 @@ void Player::play_messages_until_queue_empty(const PlayOptions & options)
   }
 }
 
+rclcpp::QoS Player::publisher_qos_for_topic(const std::string & topic_name)
+{
+  auto qos_it = topic_qos_profile_overrides_.find(topic_name);
+  if (qos_it != topic_qos_profile_overrides_.end()) {
+    ROSBAG2_TRANSPORT_LOG_INFO_STREAM("Overriding QoS profile for topic " << topic_name);
+    return Rosbag2QoS{qos_it->second};
+  } else {
+    return Rosbag2QoS{};
+  }
+}
```

Comment:
nit: instead of putting in an else block it's better to make this final fall-through case stand alone
```suggestion
  } 
  return Rosbag2QoS{};
}
```